airbyte-integrations/connectors/source-netsuite-odbc/source_netsuite_odbc/streams.py

```diff
@@ -0,0 +1,424 @@
+
+from airbyte_cdk.sources.streams import Stream, IncrementalMixin
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union, Final
+from airbyte_cdk.models import AirbyteMessage, AirbyteStream, SyncMode
+from datetime import date
+from airbyte_cdk.models import (
+    AirbyteMessage,
+    AirbyteStateMessage,
+    Type,
+    AirbyteStateType,
+    AirbyteStreamState,
+    StreamDescriptor,
+    FailureType,
+)
+import pyodbc
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from dateutil import parser
+from .errors import NETSUITE_CONNECTION_EXPIRED_FAILURE
+from .odbc_utils import NetsuiteODBCCursorConstructor
+
+# A stream's read method can return one of the following types:
+# Mapping[str, Any]: The content of an AirbyteRecordMessage
+# AirbyteMessage: An AirbyteMessage. Could be of any type
+StreamData = Union[Mapping[str, Any], AirbyteMessage]
+
+NETSUITE_PAGINATION_INTERVAL: Final[int] = 10000
+EARLIEST_DATE: Final[str] = date(2020, 1, 1)
+YEARS_FORWARD_LOOKING: Final[int] = 1
```

Comment:
is the 1 year forward looking a hard limit or could someone want to look further ahead?

Reply:
There is a world where someone MIGHT want to look further ahead, but I don't see that happening personally.  We add the one year look forward because its reasonable for an accountant to potentially do some forward looking work (e.g. record some transactions in January 2024 in Dec 2023), but further than a year out would be extremely surprising to me.

I'm confident that this can be a hard limit for now, and we can add configuration in the future if needed.